app/code/Magento/InventoryConfigurableProduct/Ui/DataProvider/Product/Form/Modifier/InventoryConfigurablePanel.php

```diff
@@ -55,10 +79,14 @@ private function getQuantityContainerConfig(): array
      */
     public function modifyMeta(array $meta)
     {
-        $children = 'children';
-        $meta[ConfigurablePanel::GROUP_CONFIGURABLE][$children]
-            [ConfigurablePanel::CONFIGURABLE_MATRIX][$children]
-            [static::RECORD][$children][static::QUANTITY_CONTAINER] = $this->getQuantityContainerConfig();
+        if ($this->allowedProductTypes->isAllowedProductType($this->locator->getProduct())) {
```

Comment:
Why do we need to use this code for products different from configurable?

Reply:
Because simple, virtual and downloadable product can be converted to configurable.

Reply:
After moving `InventoryConfigurablePanel ` we can remove this double check